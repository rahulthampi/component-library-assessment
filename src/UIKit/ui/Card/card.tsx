import * as React from 'react';

import { cn } from '@@/lib/utils';
import { Slot } from '@radix-ui/react-slot';
import { type VariantProps, cva } from 'class-variance-authority';

const cardVariants = cva(
  'flex rounded border bg-background transition-colors outline-none focus:border-border-ring',
  {
    variants: {
      variant: {
        outlined: 'border-border hover:bg-neutral',
        elevated: 'border-transparent shadow transition-shadow hover:shadow-md',
        ghost: 'border-none hover:bg-neutral',
      },
      size: {
        inset: 'p-0',
        sm: 'p-2',
        md: 'p-4',
        lg: 'p-6',
      },
    },
    defaultVariants: {
      variant: 'outlined',
      size: 'md',
    },
  },
);

interface CardProps
  extends React.HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof cardVariants> {
  asChild?: boolean;
}

const Card = React.forwardRef<HTMLDivElement, CardProps>(
  ({ variant, size, className, asChild, ...props }, ref) => {
    const Comp = asChild ? Slot : 'div';

    return <Comp ref={ref} className={cn(cardVariants({ variant, size, className }))} {...props} />;
  },
);
Card.displayName = 'Card';

export type { CardProps };

export { Card };
