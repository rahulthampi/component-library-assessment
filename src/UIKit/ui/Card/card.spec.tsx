import { render, screen } from '@testing-library/react';

import { Card } from './card';

describe('Card', () => {
  it('renders a default card with the "outlined" variant and "md" size', () => {
    const rendered = render(<Card>Hello, World!</Card>);

    expect(rendered).toMatchSnapshot();

    const card = screen.getByText('Hello, World!');

    expect(card).toBeInTheDocument();
    expect(card).toHaveClass('border-border hover:bg-neutral p-4');
  });

  it('renders a card with a different variant and size', () => {
    const rendered = render(
      <Card variant="elevated" size="lg">
        Large Card
      </Card>,
    );

    expect(rendered).toMatchSnapshot();

    const card = screen.getByText('Large Card');

    expect(card).toBeInTheDocument();
    expect(card).toHaveClass('border-transparent shadow transition-shadow hover:shadow-md p-6');
  });

  it('renders as a child component', () => {
    const rendered = render(
      <Card asChild>
        <span>Child Component</span>
      </Card>,
    );

    expect(rendered).toMatchSnapshot();

    const childComponent = screen.getByText('Child Component');

    expect(childComponent).toBeInTheDocument();
    expect(childComponent).toHaveClass('border-border hover:bg-neutral p-4');
  });
});
