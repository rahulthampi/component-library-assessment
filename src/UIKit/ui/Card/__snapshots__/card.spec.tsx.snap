// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders a card with a different variant and size 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex rounded border bg-background outline-none focus:border-border-ring border-transparent shadow transition-shadow hover:shadow-md p-6"
      >
        Large Card
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex rounded border bg-background outline-none focus:border-border-ring border-transparent shadow transition-shadow hover:shadow-md p-6"
    >
      Large Card
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card renders a default card with the "outlined" variant and "md" size 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex rounded border bg-background transition-colors outline-none focus:border-border-ring border-border hover:bg-neutral p-4"
      >
        Hello, World!
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex rounded border bg-background transition-colors outline-none focus:border-border-ring border-border hover:bg-neutral p-4"
    >
      Hello, World!
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card renders as a child component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="flex rounded border bg-background transition-colors outline-none focus:border-border-ring border-border hover:bg-neutral p-4"
      >
        Child Component
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="flex rounded border bg-background transition-colors outline-none focus:border-border-ring border-border hover:bg-neutral p-4"
    >
      Child Component
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
